File: backend/src/main/java/com/davnig/invoicemanagementapi/model/entity/Invoice.java
package com.davnig.invoicemanagementapi.model.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.sql.Date;

@Entity
@Table(name = "t_invoice")
@Getter
@Setter
public class Invoice {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "invoice-id-gen")
    @SequenceGenerator(name = "invoice-id-gen", sequenceName = "t_invoice_id_seq", allocationSize = 1)
    private int id;
    private int number;
    private Date date;
    @ManyToOne
    @JoinColumn(name = "id_vendor")
    private TaxableSubject vendor;
    @ManyToOne
    @JoinColumn(name = "id_client")
    private Client client;

}

File: backend/src/main/kotlin/com/davnig/invoicemanagementapi/model/entity/Invoice.kt
package com.davnig.invoicemanagementapi.model.entity

import jakarta.persistence.*
import java.sql.Timestamp

@Entity
@Table(name = "t_invoice")
class Invoice {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "invoice-id-gen")
    @SequenceGenerator(name = "invoice-id-gen", sequenceName = "t_invoice_id_seq", allocationSize = 1)
    var id = 0
    var number = 0

    @Column(name = "emission_date")
    var date: Timestamp? = null
    val idVendor = 0
    val idClient = 0

}